setup.py

```diff
@@ -19,7 +19,7 @@
     ],
     "lint": [
         "black>=23.3.0,<24",  # Auto-formatter and linter
-        "mypy>=0.991,<1",  # Static type analyzer
+        "mypy>=0.991,<2",  # Static type analyzer
```

Comment:
0.991 lower bound is for another use case?

Reply:
py-evm still seems to be pinned to a lower version of typing extensions than the rest of the world.
this fails in ci/cd.
but for some reason, locally i can install mypy 1.0 range... so i have no clue...

so i kept it 0.991 so itll work in ci/cd i guess.